cairo-vm-cli/src/main.rs

```diff
@@ -43,43 +43,61 @@ fn validate_layout(value: &str) -> Result<(), String> {
     }
 }
 
-fn main() -> Result<(), CairoRunError> {
+#[derive(Debug, Error)]
+enum Error {
+    #[error("Failed to interact with the file system")]
+    IO(#[from] std::io::Error),
+    #[error("The cairo program execution failed")]
+    Runner(#[from] CairoRunError),
+    #[error(transparent)]
+    EncodeTrace(#[from] EncodeTraceError),
+}
+
+fn main() -> Result<(), Error> {
     let args = Args::parse();
     let trace_enabled = args.trace_file.is_some();
     let mut hint_executor = BuiltinHintProcessor::new_empty();
     let cairo_run_config = cairo_run::CairoRunConfig {
         entrypoint: &args.entrypoint,
         trace_enabled,
-        print_output: args.print_output,
         layout: &args.layout,
         proof_mode: args.proof_mode,
         secure_run: args.secure_run,
     };
-    let cairo_runner =
-        match cairo_run::cairo_run(&args.filename, &cairo_run_config, &mut hint_executor) {
-            Ok(runner) => runner,
-            Err(error) => {
-                println!("{error}");
-                return Err(error);
-            }
-        };
+
+    let program_content = std::fs::read(args.filename).map_err(|e| Error::IO(e))?;
+
+    let (cairo_runner, opt_output) = match cairo_run::cairo_run(
+        &program_content,
+        &cairo_run_config,
+        &mut hint_executor,
+        args.print_output,
+    ) {
+        Ok(runner) => runner,
+        Err(error) => {
+            println!("{error}");
+            return Err(Error::Runner(error));
+        }
+    };
+
+    if args.print_output {
+        if let Some(output) = opt_output {
+            print!("{}", output)
+        }
```

Comment:
The `"Program Output:"` header is missing here

Reply:
done